pkl-core/src/main/java/org/pkl/core/stdlib/base/YamlRendererNodes.java

```diff
@@ -278,8 +278,24 @@ protected void visitEntryKey(Object key, boolean isFirst) {
         startNewLine();
       }
 
-      if (key instanceof String string) {
-        emitter.emit(string, currIndent, true);
+      if (key instanceof String stringKey) {
+        emitter.emit(stringKey, currIndent, true);
+        builder.append(':');
+        return;
+      } else if (key instanceof Long longKey) {
+        emitter.emit(longKey);
+        builder.append(':');
+        return;
+      } else if (key instanceof Double doubleKey) {
```

Comment:
I am not sure with that, but what about Int(eger)? And why not just use Number instead?

Reply:
Pkl's `Int` types are all represented by Java `long` internally (likewise, Pkl `Float` is always Java `double`). In this case, `emitter` has two method overloads `emit(long)` and `emit(double)` so having a single case for `Number` doesn't work here.